### PR TITLE
feat(helm): make cache TTL configurable

### LINKED ISSUE #2797: Helm chart: make cache TTL configurable
I use Helm to maintain a Trivy server. I would like to setup a Redis cache and I want to set a cache TTL. The chart does not seem to support that at the moment.

### PR DESCRIPTION
**Pull request by elchenberg**
## Description

Enables users to configure the cache TTL with a new helm value.

## Related issues
- Close #2797 

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

Signed-off-by: elchenberg <elchenberg@users.noreply.github.com>

### CODE REVIEW

**Comment by elchenberg** [comment]
Regarding the CLA: The CLA assistant asks me to sign a "SAP Individual Contributor License Agreement" (SAP as in the german software company). That does not seem right.

Regarding the PR title: I am not sure what type of PR this is. A feature? A bugfix? A chore? Or something else?

**Comment by knqyf263** [comment]
>Regarding the PR title: I am not sure what type of PR this is. A feature? A bugfix? A chore? Or something else?

It is a feature.

@itaysk Can you take a look?
>Regarding the CLA: The CLA assistant asks me to sign a "SAP Individual Contributor License Agreement" (SAP as in the german software company). That does not seem right.

**Comment by knqyf263** [comment]
@elchenberg Seems like a bug of cla-assistant. We'll look into it later, but I reverted it for now. Could you try that again?

**Comment by knqyf263** [comment]
@krol3 Can you please review it?

File: helm/trivy/templates/configmap.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ data:
   TRIVY_CACHE_DIR: "/home/scanner/.cache/trivy"
 {{- if .Values.trivy.cache.redis.enabled }}
   TRIVY_CACHE_BACKEND: {{ .Values.trivy.cache.redis.url | quote }}
+  TRIVY_CACHE_TTL: {{ .Values.trivy.cache.redis.ttl | quote }}
```
Comment (krol3) [inline helm/trivy/templates/configmap.yaml:12]: Please update the [readme- parameters section](https://github.com/aquasecurity/trivy/blob/main/helm/trivy/README.md)

| `trivy.cache.redis.ttl`               | Specify redis ttl. e.g 3600s, 24h

**Review by krol3 — CHANGES_REQUESTED** [review]
Hi @elchenberg , please only add in the readme this new parameter.

File: helm/trivy/templates/configmap.yaml (hunk 1 above)
Comment (elchenberg) [reply helm/trivy/templates/configmap.yaml:12]: Thanks, I forgot about that!

**Review by krol3 — APPROVED** [review]
LGTM

**Comment by zetaab** [comment]
@knqyf263 @elchenberg its impossible to use these new things because chart versions are not bumped when you do changes to charts. Now the only way to use newest trivy helm chart things is to clone it locally and run. If you could https://github.com/aquasecurity/trivy/blob/main/helm/trivy/Chart.yaml#L4 increase this version, everyone could use these charts quite easily.

**Comment by zetaab** [comment]
also your ci pipelines is failing see: https://github.com/aquasecurity/trivy/actions/runs/3359974186/jobs/5568645331

Screenshot 2022-11-17 at 18 00 13


it fails because the version is not increased.

**Pull request merged**